### ISSUE TITLE
Friends&Foes 3.0.0 internal changes

### ISSUE THREAD

**Issue by Faboslav** [Fourmisain/IndyPets#23]
Hello, i believe that you reference Glare in your project and since i did some internal changes in the 3.0.0 version it now throws an error, https://github.com/Faboslav/friends-and-foes/issues/267. Also apart from the crash, it would be probably good to prevent such a crash if the compat doesnt exists in any way

**Comment by Fourmisain** [comment]
Oh hi, thanks for letting me know!
Will fix the compat soon - if I haven't already, looks like the targeted classes were just moved.

> Also apart from the crash, it would be probably good to prevent such a crash if the compat doesnt exists in any way

The mixins are already non-required so they would soft-fail, but there's a piece of code that stops the current movement of the glare, which needs a reference to `GlareEntity` and pretty much the only way to make that not crash is to catch the `NoClassDefFoundError` which seems like a bad idea.

On the plus side, it'd probably have taken months to catch the broken compat otherwise.
(I have so many mods with compat code, it's basically impossible for me to test everything all the time.)
Sorry for the reports you are getting though!

**Comment by Fourmisain** [comment]
Aight, compat is fixed in 1.4.3!
Took a lot longer than I thought since the way I cycled the sitting/following/independent states didn't work with the Glare, so I had to rewrite the logic.
That one piece of code also shouldn't crash anymore (it was just an `instanceof`, I replaced it with an id check).

> I mean creating some kind of an error proof wrapper around the compat parts of the mod seems like good idea, i recently did this too, if you are interested bumblezone mod have it pretty decently handled

Will keep that in mind!
I generally prefer stuff to crash, so people tell me it's broken. Just in this case I thought it's too small of a feature to hold up the whole mod - and it did anyway, oh well... 😅
